configure

```diff
@@ -3764,16 +3764,29 @@ return f (e, argv, 0) != argv[0]  ||  f (e, argv, 1) != argv[1];
   return 0;
 }
 _ACEOF
-for ac_arg in '' -qlanglvl=extc89 -qlanglvl=ansi -std \
-	-Ae "-Aa -D_HPUX_SOURCE" "-Xc -D__EXTENSIONS__"
-do
-  CC="$ac_save_CC $ac_arg"
-  if ac_fn_c_try_compile "$LINENO"; then :
-  ac_cv_prog_cc_c89=$ac_arg
+if [[ "$CC" =~ 'ibm-clang' ]] && [[ "$CXX" =~ 'ibm-clang' ]]; then
+  for ac_arg in '' -std=extc89 -std=ansi -std \
+    -Ae "-Aa -D_HPUX_SOURCE" "-Xc -D__EXTENSIONS__"
+  do
+    CC="$ac_save_CC $ac_arg"
+    if ac_fn_c_try_compile "$LINENO"; then :
+      ac_cv_prog_cc_c89=$ac_arg
+    fi
+    rm -f core conftest.err conftest.$ac_objext
+    test "x$ac_cv_prog_cc_c89" != "xno" && break
+  done
+else
+  for ac_arg in '' -qlanglvl=extc89 -qlanglvl=ansi -std \
+    -Ae "-Aa -D_HPUX_SOURCE" "-Xc -D__EXTENSIONS__"
+  do
+    CC="$ac_save_CC $ac_arg"
+    if ac_fn_c_try_compile "$LINENO"; then :
+    ac_cv_prog_cc_c89=$ac_arg
+    fi
+    rm -f core conftest.err conftest.$ac_objext
+    test "x$ac_cv_prog_cc_c89" != "xno" && break
```

Comment:
I think there's a better way to do this, namely, `ac_arg` can be set conditionally based on `if [[ "$CC" =~ 'ibm-clang' ]] && [[ "$CXX" =~ 'ibm-clang' ]]` and then `do/done` body can be common across both.